playground/frontend/playground_components/lib/widgets/split_view/controller.dart

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import '../../constants/sizes.dart';
+
+class SplitViewController extends ChangeNotifier {
+  final bool isHorizontal;
```

Comment:
Please reuse Flutter's `enum Axis`.

Reply:
Second thoughts, there is a problem with the controller deciding on the axis.

Imagine the `SplitView` widget in a container with a finite width and an infinite height. The widget should build OK with horizontal layout taking the height of its highest row. But the widget will break with vertical layout because it cannot go `Expanded` to the infinity. (I guess, the current implementation relies on both dimension being finite, but we need not to introduce more obstacles to infinity should we ever need it).

Thus the controller and the widget tree collectively determine if the widget will break or not, and this is a bad blurring of responsibility.

Ideally `SplitViewController` should only know the ratio and not the direction. This also goes in line with Flutter's `ScrollController` which also does not know the direction and can be applied to a widget with any.

Reply:
By the way, Flutter 's widgets tend to name this property `direction`, and it's a good name for `SplitView`'s property as well.

Reply:
Changed to use https://pub.dev/packages/split_view